content/lib/2021-04-02-operating-system-interface.md

```diff
@@ -0,0 +1,225 @@
+         - CNI is [required by a container runtime](https://github.com/containernetworking/cni/blob/master/SPEC.md#overview-1), however it is assumed this is installed from `kubernetesland`. This is a paradox. 
+     - CNI assumes access to `userland` and more importantly access (and permission) to `fork(2)` and `execve(2)`
+     - CNI does not solve *how* access is managed, or acquired. 
+ - CNI does not offer a standardized way of reporting, measuring, or limiting resources in `kubernetesland`.
+ 
+### Storage (CSI)
+
+Official [Container Storage Interface](https://github.com/container-storage-interface/spec) standard.
+
+ - CSI defines the container storage API surface/shape for interfacing with block and file storage drivers in `userland`. 
+     - More information on [CSI drivers](https://kubernetes.io/blog/2019/01/15/container-storage-interface-ga/)
+     - More information on [Kubelet device registration and plugins](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/)
+ - CSI does not solve *how* to install or manage block or file storage drivers.
+ - The Kubelet plugin system does NOT solve how to install or manage Kubelet plugins from Kubernetes. 
+     - [Access to the Kubelet API server is required](https://v1-18.docs.kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#device-plugin-registration) to interact with the exposed registration service. 
+ - CSI (via the Kubelet registration endpoint and logic) can set limitations and provide status updates in `kubernetesland`.
+
+
+### Runtime (Compute) (CRI/OCI)
+
+Official [Container Runtime Interface](https://github.com/kubernetes/cri-api) API.
+Official [Open Container Interface](https://github.com/opencontainers/runtime-spec) standard.
+    - Official [Open Container Image Spec](https://github.com/opencontainers/image-spec)
+
+CRI is a implementation of OCI.
+The main difference is scope, in which case OCI also defines the [open container image spec](https://github.com/opencontainers/image-spec).
+
+
+- OCI and CRI define the container runtime API surface/shape for interfacing with container runtime "engines" from `userland`. 
+    - OCI also defines the API surface/shape and interface for packaging container images
+- Container runtimes are tightly coupled with Linux features, and therefor are tightly coupled with the operating system. 
+- Container runtimes manage permissions, storage, and networking in different ways. 
+- Container runtimes are tightly coupled with `storage` and `networking`.
+- Container runtimes do not define *how* they are installed or managed. 
+
+# The Void
+
+There are several `userland` components that are not considered from a management perspective. 
+
+Below are demonstrable questions that technically have answers.
+I hope to challenge the existing answers to these questions with DOSi.
+
+### Demonstrable Questions
```

Comment:
feel free to add more questions here that you think could have a better answer if something like DOSi was considered